app/models/miq_expression.rb

```diff
@@ -1880,7 +1878,7 @@ def visit_Arel_Nodes_SelectCore(o, collector)
   end
 
   def extract_where_values(klass, scope)
-    relation = ActiveRecord::Relation.new klass, klass.arel_table
+    relation = ActiveRecord::Relation.new klass, klass.arel_table, klass.predicate_builder
```

Comment:
@matthewd I would appreciate your looking over my changes to this method, thanks!